ffmpeg_plugin/0001-lavc-svt_hevc-add-libsvt-hevc-encoder-wrapper.patch

```diff
@@ -510,6 +568,49 @@ index 0000000000..fce27a06ea
 +
 +    EbH265ReleaseOutBuffer(&header_ptr);
 +
++    if (svt_enc->ref_frame) {
++#ifdef _WIN32
++        // To check whether the list is empty or not, and remove the input buffer
++        // which has been encoded from the encoding list.
++        first_entry = InterlockedPopEntrySList(&svt_enc->input_frame_list);
++        input_frame_entry = (input_frame_t *)first_entry;
++
++        while (input_frame_entry) {
++            if (input_frame_entry->ref_frame->pts == pkt->pts) {
++                // Unference the AVFrame oject which has been encoded completely.
++                av_frame_unref(input_frame_entry->ref_frame);
++                free(input_frame_entry);
++
++                if (QueryDepthSList(&svt_enc->tmp_input_frame_list) &&
++                        last_entry && first_entry) {
++                    // first_entry has been pushed to the last entry of tmp_input_frame_list.
++                    InterlockedPushListSListEx(&svt_enc->input_frame_list, last_entry,
```

Comment:
Seems that using `InterlockedPushListSListEx` requires at least Windows 8, and defining `NTDDI_VERSION` to at least `0x06020000`. There seems to be [`InterlockedPushListSList`](https://docs.microsoft.com/en-us/windows/win32/sync/interlockedpushlistslist) which instead requires the windows version to be higher than windows xp (Which in turn gets defined to `InterlockedPushListSListEx` if the windows version is greater than 8)
```c
libavcodec/libsvt_hevc.c: In function 'eb_encode_frame':
libavcodec/libsvt_hevc.c:501:21: error: implicit declaration of function 'InterlockedPushListSListEx'; did you mean 'RtlInterlockedPushListSListEx'? [-Werror=implicit-function-declaration]
  501 |                     InterlockedPushListSListEx(&svt_enc->input_frame_list, last_entry,
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~
      |                     RtlInterlockedPushListSListEx
```

Reply:
Thanks, @1480c1! As using InterlockedPushListSList still caused the compiling issue, I've to drop it with another way instead.